app/Actions/Statistics/Spaces.php

```diff
@@ -0,0 +1,342 @@
+				as: 'size_variants',
+				first: 'size_variants.photo_id',
+				operator: '=',
+				second: 'photos.id',
+				type: 'left'
+			)
+			->select(
+				'users.id',
+				'username',
+				DB::raw('SUM(size_variants.filesize) as size')
+			)
+			->groupBy('users.id', 'username')
+			->orderBy('users.id', 'asc')
+			->get()
+			->map(fn ($item) => [
+				'id' => intval($item->id),
+				'username' => strval($item->username),
+				'size' => intval($item->size),
+			]);
+	}
+
+	/**
+	 * Return the amount of data stored on the server (optionally for a user).
+	 *
+	 * @param int|null $owner_id
+	 *
+	 * @return Collection<int,array{type:SizeVariantType,size:int}>
+	 */
+	public function getSpacePerSizeVariantType(?int $owner_id = null): Collection
+	{
+		return DB::table('size_variants')
+			->when($owner_id !== null, fn ($query) => $query
+				->joinSub(
+					query: DB::table('photos')->select(['photos.id', 'photos.owner_id']),
+					as: 'photos',
+					first: 'photos.id',
+					operator: '=',
+					second: 'size_variants.photo_id'
+				)
+				->where('photos.owner_id', '=', $owner_id))
+			->select(
+				'size_variants.type',
+				DB::raw('SUM(size_variants.filesize) as size')
+			)
+			->groupBy('size_variants.type')
+			->orderBy('size_variants.type', 'asc')
+			->get()
+			->map(fn ($item) => [
+				'type' => SizeVariantType::from($item->type),
+				'size' => intval($item->size),
+			]);
+	}
+
+	/**
+	 * Return size statistics per album.
+	 *
+	 * @param string|null $album_id
+	 * @param int|null    $owner_id
+	 *
+	 * @return Collection<int,array{id:string,left:int,right:int,size:int}>
+	 */
+	public function getSpacePerAlbum(?string $album_id = null, ?int $owner_id = null)
+	{
+		$query = DB::table('albums')
+			->when($album_id !== null, fn ($query) => $query->where('albums.id', '=', $album_id))
+			->when($owner_id !== null, fn ($query) => $query->joinSub(
+				query: DB::table('base_albums')->select(['base_albums.id', 'base_albums.owner_id']),
+				as: 'base_albums',
+				first: 'base_albums.id',
+				operator: '=',
+				second: 'albums.id'
+			)
+				->where('base_albums.owner_id', '=', $owner_id))
+			->joinSub(
+				query: DB::table('photos'),
+				as: 'photos',
+				first: 'photos.album_id',
+				operator: '=',
+				second: 'albums.id'
+			)
+			->joinSub(
+				query: DB::table('size_variants')->select(['size_variants.id', 'size_variants.photo_id', 'size_variants.filesize']),
+				as: 'size_variants',
+				first: 'size_variants.photo_id',
+				operator: '=',
+				second: 'photos.id'
+			)
+			->select(
+				'albums.id',
+				'albums._lft',
+				'albums._rgt',
+				DB::raw('SUM(size_variants.filesize) as size'),
+			)->groupBy('albums.id')
+			->orderBy('albums._lft', 'asc');
+
+		return $query
+			->get()
+			->map(fn ($item) => [
+				'id' => strval($item->id),
+				'left' => intval($item->_lft),
+				'right' => intval($item->_rgt),
+				'size' => intval($item->size),
+			]);
+	}
+
+	/**
+	 * Same as above but with full size.
```

Comment:
I think this is right?

```suggestion
	 * Same as above but with full size (including sub-albums).
```